pilot/pkg/networking/core/route/route.go

```diff
@@ -643,6 +644,18 @@ func applyHTTPRouteDestination(
 		}
 	}
 	action.RetryPolicy = retry.ConvertPolicy(policy, consistentHash)
+	// TODO: Remove two or more releases later
+	if strings.Contains(action.RetryPolicy.RetryOn, "reset-before-request") {
```

Comment:
I do not think this is needed. Envoy will just ignore this even if we send to older proxies

Reply:
If that true, it would be quite good.

Reply:
Not sure why the `checkHeades` will validate the retryON
```
  if (config_->strict_check_headers_ != nullptr) {
    for (const auto& header : *config_->strict_check_headers_) {
      const auto res = FilterUtility::StrictHeaderChecker::checkHeader(headers, header);
      if (!res.valid_) {
        callbacks_->streamInfo().setResponseFlag(
            StreamInfo::CoreResponseFlag::InvalidEnvoyRequestHeaders);
        const std::string body = fmt::format("invalid header '{}' with value '{}'",
                                             std::string(res.entry_->key().getStringView()),
                                             std::string(res.entry_->value().getStringView()));
        const std::string details =
            absl::StrCat(StreamInfo::ResponseCodeDetails::get().InvalidEnvoyRequestHeaders, "{",
                         StringUtil::replaceAllEmptySpace(res.entry_->key().getStringView()), "}");
        callbacks_->sendLocalReply(Http::Code::BadRequest, body, nullptr, absl::nullopt, details);
        return Http::FilterHeadersStatus::StopIteration;
      }
    }
  }
```

Reply:
Envoy will only validate if it is passed via x-envoy-retry-on header with strict header matching on.

Reply:
got it